src/screens/Survey/sagas.js

```diff
@@ -0,0 +1,33 @@
+import { call, put, select, takeLatest } from 'redux-saga/effects';
+import {selectUserReference, writeResponsesToUserRef} from './lib/api'
+import {storeQuestionKey, writeSuccess, writeFailure, SURVEY_NAMESPACE} from './redux/actions/Actions_question'
+import * as actionType from './redux/actions/ActionType';
+import {AUTH_NAMESPACE} from '../../auth/redux';
+
+const getarrayOfObjects = state => state[SURVEY_NAMESPACE].questionResponses;
```

Comment:
I think it would be better to keep these in the reducer so that the structure of the reducer doesn't leak out. In my experience, using selectors as a layer of abstraction, where external modules only interact with the reducer using action creators and selectors, makes it much easier to refactor, because changing the structure of the reducer only requires changing those functions, rather than chasing down all the usages in all the different files

Reply:
Yup! You are absolutely right. I removed it from there!

Reply:
Also, about the font family, that error doesn't show on my emulator, but I have an android. I found a way to add the font for ios as well, and I hope that works. I'll try to download an iOs emulator to check that as well.